app/code/Magento/Inventory/Model/Stock/Validator/SalesChannelsValidator.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Inventory\Model\Stock\Validator;
+
+use Magento\Framework\Validation\ValidationResult;
+use Magento\Framework\Validation\ValidationResultFactory;
+use Magento\InventoryApi\Api\Data\StockInterface;
+use Magento\InventoryApi\Model\StockValidatorInterface;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+
+/**
+ * Check that sales channels are correct
+ */
+class SalesChannelsValidator implements StockValidatorInterface
```

Comment:
Looks like we need to move this validator to `InventorySales` module due to  `InventorySales`  introduces `SalesChannel` concept